src/numeric-types.c

```diff
@@ -112,11 +214,25 @@ DEFINE_NUMERIC_WITH_BYTESWAP (double_be, double, gint32,  GINT64,  BE)
 
 #if HAVE_LIBDFP
 extern void register_printf_dfp (void);
+#endif
 
 __attribute__ ((constructor))
 static void
 numeric_types_init (void)
 {
+    // register all GLib numerical types
+    numeric_type_register_static (G_TYPE_CHAR,   "char",   sizeof (gchar),   G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_UCHAR,  "uchar",  sizeof (guchar),  G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_INT,    "int",    sizeof (gint),    G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_UINT,   "uint",   sizeof (guint),   G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_LONG,   "long",   sizeof (glong),   G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_ULONG,  "ulong",  sizeof (gulong),  G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_INT64,  "int64",  sizeof (gint64),  G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_UINT64, "uint64", sizeof (guint64), G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_FLOAT,  "float",  sizeof (gfloat),  G_BYTE_ORDER);
+    numeric_type_register_static (G_TYPE_DOUBLE, "double", sizeof (gdouble), G_BYTE_ORDER);
```

Comment:
The issue here is that we cannot redefine these types. We can either have a numeric boxed counterpart or look into switching from boxed types to fundamental types for our definitions.

The quickfix would be to define these "fundamental" types lazily in `numeric_get_type_info`.

Reply:
2 quick points: Would life be made easier if not using macros but hard code and/or generate the definitions with a code generator?

I remember the get_type() functions always generate the type the first time. Is that what you gain by moving to a fundamental type? If adding a fundamental type. I would suggest considering GNumericScalar as an umbrella type for all of the types.

Reply:
It's actually doing just that! `G_DEFINE_BOXED_TYPE_WITH_CODE` inserts arbitrary code inside the `_get_type` definition allowing us to perform any kind of initialization and registration of transformations.

The issue I'm facing with GLib fundamental types is that we cannot redefine that piece of code and thus registering the `NumericTypeInfo` counterpart we will need for introspection and byteswapping.